File: sample/src/main/java/com/mandria/android/mvp/example/di/modules/PresenterModule.java
package com.mandria.android.mvp.example.di.modules;

import com.mandria.android.mvp.Presenter;
import com.mandria.android.mvp.example.di.annotations.PresenterKey;
import com.mandria.android.mvp.example.mvp.presenters.MainPresenter;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;

@Module
public abstract class PresenterModule {

    @Binds
    @IntoMap
    @PresenterKey(MainPresenter.class)
    abstract Presenter bindMainPresenter(MainPresenter presenter);
}

File: demo_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/kotlin/example/di/PresenterModule.kt
package com.mandria.android.mvp.kotlin.example.di

import com.mandria.android.mvp.Presenter
import com.mandria.android.mvp.kotlin.example.mvp.presenter.MainPresenter
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap


@Module
abstract class PresenterModule {

    @Binds
    @IntoMap
    @PresenterKey(MainPresenter::class)
    abstract fun bindDatabasePresenter(presenter: MainPresenter): Presenter<*>
}